corehq/apps/reports/filters/users.py

```diff
@@ -364,60 +373,13 @@ def user_es_query(cls, domain, mobile_user_and_group_slugs):
             else:
                 return q.filter(id_filter)
 
-    @classmethod
-    def pull_users_and_groups(cls, domain, mobile_user_and_group_slugs,
```

Comment:
Might want to check this again.
1. I vaguely remember that we used to pull groups for selected users as well and then the users under those groups to then fetch cases that are owned by the group/users under it. Is that still happening somewhere? this was probably because for case sharing between groups? Sorry i don't remember exactly where this was.
Related to this the check now is just for selected users using `_verify_users_are_accessible ` and not for user_ids generated/added by other filters. We should confirm if user_ids are not getting added by other filter like group. Because if thats the case one would be able to view cases/forms submitted by inaccessible user under a group.

Reply:
Restricted users can't filter by group, so I don't think that should be an issue.  They only see users and locations.  Selected groups are taken care of here: https://github.com/dimagi/commcare-hq/pull/18544/files#diff-07031c94878b608626b437e634d152ddR365

This method returns a bunch of different things, but only one of them was in-use, so I swapped those usages out with `user_es_query` for simplicity.